vms/platformvm/service.go

```diff
@@ -510,6 +511,73 @@ func (s *Service) GetUTXOs(_ *http.Request, args *api.GetUTXOsArgs, response *ap
 	return nil
 }
 
+// GetSubnetArgs are the arguments to GetSubnet
+type GetSubnetArgs struct {
+	// ID of the subnet to retrieve information about
+	SubnetID ids.ID `json:"subnetID"`
+}
+
+// GetSubnetResponse is the response from calling GetSubnet
+type GetSubnetResponse struct {
+	// whether it is permissioned or not
+	IsPermissioned bool `json:"isPermissioned"`
+	// subnet auth information for a permissioned subnet
+	ControlKeys []string    `json:"controlKeys"`
+	Threshold   json.Uint32 `json:"threshold"`
+	Locktime    json.Uint64 `json:"locktime"`
+	// subnet transformation tx ID for a permissionless subnet
+	SubnetTransformationTxID ids.ID `json:"subnetTransformationTxID"`
```

Comment:
q: I wonder if we should return the whole parameters here instead of simply the txID? In such a casa I would return also Primary network config when primary network ID is specified.
I am not sure we expose this config via API currently and it'd be a more uniform handling of subnets and primary network

Reply:
probably it is best to return formatted params instead of requiring the user to parse the tx. if @StephenButtolph agrees will make the change. also, regarding primary network params.

Reply:
I'd be willing to mrege this PR as is. If we wanted to return more information about permissionless subnets I wouldn't necessarily object.